girder/utility/gridfs_assetstore_adapter.py

```diff
@@ -65,6 +65,7 @@ def validateInfo(doc):
                 ('uuid', pymongo.ASCENDING),
                 ('n', pymongo.ASCENDING)
             ], unique=True)
+            chunkColl.create_index([('n', pymongo.ASCENDING)])
```

Comment:
It turns out that somehow I had dropped the indices from the GridFS database.  I then uploaded a large file to it, and it wouldn't download due to the lack of any index on `n`.  The addition of the `n`-only index was a red herring.  We really just need to ensure the index exists when we attach to the database (in the `__init__` method), though we should still do so in the `validate` method as well.